packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -333,18 +335,18 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
                     this.logger.sendErrorEvent({ eventName: "TreeLatest_SecondCall", hasClaims: !!tokenFetchOptions.claims });
                 }
 
+                // Choose min of host specified limits or driver specified limits.
+                const maxSnapshotSizeLimit = 25000000; // 25 MB
+                const maxSnapshotFetchTimeout = 30000; // 30 sec
                 const snapshotOptions: ISnapshotOptions = {
                     deltas: 1,
                     channels: 1,
                     blobs: 2,
                     ...this.hostPolicy.snapshotOptions,
+                    mds: this.hostPolicy.snapshotOptions?.mds ? Math.min(this.hostPolicy.snapshotOptions.mds, maxSnapshotSizeLimit) : maxSnapshotSizeLimit,
+                    timeout: this.hostPolicy.snapshotOptions?.timeout ? Math.min(this.hostPolicy.snapshotOptions.timeout, maxSnapshotFetchTimeout) : maxSnapshotFetchTimeout,
                 };
 
-                // No limit on size of snapshot, as otherwise we fail all clients to summarize
-                if (this.hostPolicy.summarizerClient) {
```

Comment:
This does not look right. The limits client applied to original snapshot can't be applied to summarizer client in the same way. Otherwise we can easily run into situation where client started with 999K snapshot, but summarizer failed to load at 1001K snapshot, and now client keeps going pushing a ton of ops without any chance to summarize.
It's a balancing act between ignoring host requirements (summarizer can join much later when file is bigger) and not summarizing file at all.

Reply:
How come this is a special situation for summarizer? If a normal client failed to load a big snapshot, then the summarizer client would also fail? (as you said you ran out of memory). But I agree that snapshot client needs to download the full snapshot.

Reply:
Note that summarizer client can boot 20 hours later because some other client (who was summarizer) left.
So not, it's possible that main client succeeded to load at 999K (with limit of 1M), but summarizer would fail at 1001K if limits are there. Unfortunately we have no choice other than let summarizer in, but we can clearly optimize bandwidth over speed for summarizer (i..e not load it with blobs in such case). But we can't fail it